packages/rum-core/src/error-logging/error-logging.js

```diff
@@ -171,22 +171,24 @@ class ErrorLogging {
   }
 
   logPromiseEvent(promiseRejectionEvent) {
-    const prefix = 'Unhandled promise rejection: '
-    const { reason } = promiseRejectionEvent
+    const prefix = 'Uncaught (in promise) '
+    const { reason = '<no reason specified>' } = promiseRejectionEvent
+    let errorEvent
 
-    if (reason == null) {
-      this.logError(prefix + '<no reason specified>')
-    } else if (typeof reason.message === 'string') {
+    if (typeof reason.name === 'string' && typeof reason.message === 'string') {
       /**
        * Promise is rejected with an error or error like object
        */
-      this.logError({
-        message: prefix + reason.message,
-        stack: reason.stack ? reason.stack : null
-      })
-    } else if (typeof reason !== 'object') {
-      this.logError(prefix + reason)
+      errorEvent = {
+        error: reason,
+        message: `${prefix}${reason.name}: ${reason.message}`
+      }
+    } else {
+      errorEvent = {
+        message: `${prefix}${reason}`
```

Comment:
If we are going to support for all reason types, there should be a normalisation for these types.  Having `[object Object]` is not helpful.